### PR TITLE
Add integration for editing dns records

### LINKED ISSUE #331: Add record editing UI + integration
Probably can just reuse UI from creation, just prepopulate values and then call an appropriate function from records flow

### PR DESCRIPTION
**Pull request by Myrfion**
## What it changes
- creates a new page by editing the DNS record (I changed the file name to match the new naming)
- updates the form component for DNS records so it supports editing 
- makes server-side integration with DNS records workflow and its API 

closes #331

### CODE REVIEW

**Comment by Eakam1007** [comment]
Seems like tests are failing due to the same error as the Type-Check error

**Comment by humphd** [comment]
Can you rebase and update your code to use the new naming/layout that we landed in https://github.com/DevelopingSpace/starchart/pull/349?

**Review by humphd — CHANGES_REQUESTED** [review]
I tested this locally, and had to make the following changes:

```diff
diff --git a/app/routes/__index/domains/$dnsRecordId.tsx b/app/routes/__index/domains/$dnsRecordId.tsx
index bbf83d9..b6398cf 100644
--- a/app/routes/__index/domains/$dnsRecordId.tsx
+++ b/app/routes/__index/domains/$dnsRecordId.tsx
@@ -5,7 +5,7 @@ import { z } from 'zod';
 import { parseFormSafe } from 'zodix';
 import type { ActionArgs, LoaderArgs } from '@remix-run/node';
 import DnsRecordForm from '~/components/dns-record/form';
-import { updateDnsRequest } from '~/queues/dns/dns-flow.server';
+import { updateDnsRequest } from '~/queues/dns/update-record-flow.server';
 import { requireUser } from '~/session.server';
 import { getRecordById } from '~/models/record.server';

@@ -55,7 +55,7 @@ export const action = async ({ request }: ActionArgs) => {
     id: Number(data.id),
     username: user.username,
     type: data.type,
-    name: data.name,
+    subdomain: data.name,
     value: data.value,
   });
```

You should also rebase, since a lot has changed since you wrote this code.

**Comment by Myrfion** [comment]
@humphd updated to new terminology and updated terminology inside of the table component itself, except the filename, because I am not sure if it is a good idea to do it in this PR (it would be a lot of changes)

File: app/components/dns-record/dns-record-name.tsx [hunk 1]
```diff
@@ -1,20 +1,19 @@
 import { Flex, Text } from '@chakra-ui/react';
 
 interface DnsRecordNameProps {
-  name: string;
+  subdomain: string;
+  basedomain: string;
```
Comment (SerpentBytes) [inline app/components/dns-record/dns-record-name.tsx:5]: I would use camelCase for `basedomain`.

File: app/components/domains-table.tsx [hunk 2]
```diff
@@ -46,18 +47,18 @@ export default function DomainsTable(props: DomainsTableProps) {
...
       position: 'bottom-right',
       status: 'success',
     });
   }
 
-  function renderDomainStatus(action: RecordStatus) {
+  function renderDnsRecordStatus(action: RecordStatus) {
     if (action === 'active') {
       return (
         <Tooltip label="Domain is live">
```
Comment (SerpentBytes) [inline app/components/domains-table.tsx:64]: Should the `label` begin with "Domain..." or "Record..."

File: app/components/domains-table.tsx [hunk 3]
```diff
@@ -140,28 +141,31 @@ export default function DomainsTable(props: DomainsTableProps) {
...
                           <Flex justifyContent="space-between" alignItems="center">
-                            <DnsRecordName name={domain.subdomain} />
+                            <DnsRecordName
+                              subdomain={dnsRecord.subdomain}
+                              basedomain={userBaseDomain}
+                            />
                             <Tooltip label="Copy name to clipboard">
                               <IconButton
                                 icon={<CopyIcon color="black" boxSize="5" />}
                                 aria-label="Refresh domain"
```
Comment (SerpentBytes) [inline app/components/domains-table.tsx:154]: typo in `aria-label`

File: app/components/dns-record/form.tsx [hunk 4]
```diff
@@ -10,23 +10,31 @@ import {
...
 import { useUser } from '~/utils';
 import FormField from './form-field';
+import { useMemo } from 'react';
+
+type FormMode = 'CREATE' | 'EDIT';
 
 interface DnsRecordFormProps {
+  mode: FormMode;
   typeError?: string; // Error for 'Type' field
+  defaultDnsRecord?: Record;
```
Comment (humphd) [inline app/components/dns-record/form.tsx:23]: What does `default` mean here? Can this not simply be `record` or `dnsRecord`?

File: app/components/domains-table.tsx [hunk 5]
```diff
@@ -46,18 +49,18 @@ export default function DomainsTable(props: DomainsTableProps) {
...
   } = useDisclosure();
-  const [domainToDelete, setDomainToDelete] = useState<Record | undefined>();
+  const [dnsRecordToDelete, setDnsRecordToDelete] = useState<Record | undefined>();
 
-  function onCopyNameToClipboard(name: string) {
-    navigator.clipboard.writeText(name);
+  function onCopyNameToClipboard(subdomain: string) {
+    navigator.clipboard.writeText(subdomain);
     toast({
       title: 'Name was copied to clipboard',
```
Comment (humphd) [inline app/components/domains-table.tsx:57]: `Subdomain was copied...`

**Review by humphd — CHANGES_REQUESTED** [review]
Looks really good.  A few things

**Review by humphd — CHANGES_REQUESTED** [review]
Can you also change the table heading `Name` to `Subdomain` as well on line 123 of `app/components/domains-table.tsx`

**Review by humphd — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Review by SerpentBytes — APPROVED** [review]

**Pull request merged**